src/pages/index.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useState, useEffect, useContext } from 'react';
+import Head from 'next/head';
+import File from '../interfaces/file';
+import styles from '../styles/Home.module.css';
+import Table from '../components/Table';
+import Button from '../components/Button';
+import DragAndDropFile from '../components/DragAndDropFile';
+import { filesLoaded, fileAdded, fileRemoved } from '../store/fileHistory';
+import StoreContext from '../contexts/storeContext';
+
+export default function Home() {
+  const fileHistory = useContext(StoreContext);
+  const [files, setFiles] = useState<File[] | []>([]);
+  const [droppedFile, setDroppedFile] = useState<File | null>(null);
+  const [isDisabled, setIsDisabled] = useState(false);
+
+  useEffect(() => {
+    fileHistory.dispatch(filesLoaded());
+    setFiles(fileHistory.getState());
+  }, []);
+
+  const onChangeInputFile = (e: any) => {
+    setDroppedFile(e.target.files[0]);
+  };
+
+  const handleDragEnter = (e: any) => {
+    e.preventDefault();
+    if (e.target?.files?.[0]) {
+      setDroppedFile(e.target.files[0]);
+    }
+  };
+
+  const handleSaveFile = () => {
+    const t: any = {
+      id: Date.now(),
+      name: droppedFile?.name,
+      lastModified: droppedFile?.lastModified,
+      size: droppedFile?.size,
+      type: droppedFile?.type,
+    };
+    let previousFiles: any = [];
+    if (localStorage.getItem(`file-history`)) {
+      previousFiles = previousFiles.concat(
+        localStorage.getItem(`file-history`),
+      );
+    }
+
+    const updateFileHistory = previousFiles.concat(JSON.stringify(t));
+
+    localStorage.setItem(`file-history`, updateFileHistory);
+    fileHistory.dispatch(fileAdded(t));
+    setFiles(fileHistory.getState());
+    setIsDisabled(true);
+  };
+
+  const handleResetFile = () => {
+    setDroppedFile(null);
+    setIsDisabled(false);
+  };
+
+  const handleClickButton = (id: number) => {
+    let f: any = `[${localStorage.getItem(`file-history`)}]`;
+    f = JSON.parse(f);
```

Comment:
Variables should have meaningful names.

Reply:
"meaningful names" is not biggest problem in current place, I would recommend to read article about object shapes in V8, references:
- [JavaScript engine fundamentals: Shapes and Inline Caches](https://mathiasbynens.be/notes/shapes-ics)
- [The story of a V8 performance cliff in React](https://v8.dev/blog/react-cliff)

long story short - don't use `any` and don't change variable type